src/backend/opencl/magma/magma_blas_clblas.h

```diff
@@ -0,0 +1,74 @@
+/*******************************************************
+ * Copyright (c) 2014, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#pragma once
+
+#include <defines.hpp>
+
+#include <clBLAS.h>
+#include <err_clblas.hpp>
+
+// Convert MAGMA constants to clBLAS constants
+clblasOrder          clblas_order_const( magma_order_t order );
+clblasTranspose      clblas_trans_const( magma_trans_t trans );
+clblasUplo           clblas_uplo_const ( magma_uplo_t  uplo  );
+clblasDiag           clblas_diag_const ( magma_diag_t  diag  );
+clblasSide           clblas_side_const ( magma_side_t  side  );
+
+// Error checking
+#define OPENCL_BLAS_CHECK CLBLAS_CHECK
+
+// Transposing
+#define OPENCL_BLAS_TRANS_TYPE clblasTranspose // the type
```

Comment:
Also an unscoped enum doesn't seem so easy to implement: the clBLAS wrapper already takes clBLAS types (and thus unscoped enums), but the CLBlast wrapper takes CLBlast types and thus scoped enums. Not sure how to do this nicely without changing some things to the function interfaces and adding some casts. Perhaps you could clarify what you had in mind?

I did change the types from *_TYPE into *_T.